tests/defaults_list/test_defaults_tree.py

```diff
@@ -831,6 +831,63 @@ def test_legacy_hydra_overrides_from_primary_config_2(
     assert "Invalid overriding of hydra/output:" in recwarn.list[1].message.args[0]
 
 
+@mark.parametrize(
+    "config_name,overrides,expected",
+    [
+        param(
+            "legacy_override_hydra_wrong_order",
+            [],
+            DefaultsTreeNode(
+                node=VirtualRoot(),
+                children=[
+                    DefaultsTreeNode(
+                        node=ConfigDefault(path="hydra/config"),
+                        children=[
+                            GroupDefault(group="help", value="custom1"),
+                            GroupDefault(group="output", value="default"),
+                            ConfigDefault(path="_self_"),
+                        ],
+                    ),
+                    DefaultsTreeNode(
+                        node=ConfigDefault(path="legacy_override_hydra_wrong_order"),
+                        children=[
+                            GroupDefault(group="group1", value="file1"),
+                            ConfigDefault(path="_self_"),
+                        ],
+                    ),
+                ],
+            ),
+            id="legacy_override_hydra_wrong_order",
+        ),
+    ],
+)
+def test_legacy_hydra_overrides_from_primary_config_3(
+    config_name: str, overrides: List[str], expected: DefaultsTreeNode, recwarn: Any
+) -> None:
+    """
+    Override two Hydra config groups using legacy notation
+    """
+
+    _test_defaults_tree_impl(
+        config_name=config_name,
+        input_overrides=overrides,
+        expected=expected,
+        prepend_hydra=True,
+    )
+
+    assert len(recwarn) == 2
+    assert "Invalid overriding of hydra/help:" in recwarn.list[0].message.args[0]
+    assert (
+        dedent(
+            """\
+            Legacy override 'hydra/help : custom1' is defined before 'group1 : file1'.
+            Overrides should be at the end of the defaults list.
+            """
+        )
+        in recwarn.list[1].message.args[0]
+    )
+
```

Comment:
Instead of producing two warnings, we can produce just the first one.

In the area where we check the order, we can treat hydra overrides as if they are prefixed by `override`, which means they will just produce the first warning.

We can also update the first warning to indicate that overrides should also be at the end of the defaults list if it fits nicely.

What do you think?

Reply:
> Instead of producing two warnings, we can produce just the first one.
> ...
> We can also update the first warning to indicate that overrides should also be at the end of the defaults list if it fits nicely.

Sounds good. Done in https://github.com/facebookresearch/hydra/pull/1750/commits/96cca574b07ef027e7938c442a311af9b749cede.

> In the area where we check the order, we can treat hydra overrides as if they are prefixed by `override`, which means they will just produce the first warning.

I still think we need to track legacy overrides specially -- normal overrides (with the `override` prefix) will cause `ConfigCompositionException` if they are out of order, and we want to prevent this for the legacy overrides.